src/main/java/htsjdk/tribble/annotation/Strand.java

```diff
@@ -24,29 +24,116 @@
 package htsjdk.tribble.annotation;
 
 /**
- * Enum for strand, which can be encoded as string
+ * Enum for strand, which can be encoded as a string
  */
 public enum Strand {
-    POSITIVE("+"), NEGATIVE("-"), NONE("!");  // not really sure what we should do for the NONE Enum
 
     /**
-     * How we represent the strand information as text
+     * Represents the positive or forward strand.
      */
-    private String encoding;
-    Strand(String str) {
-        encoding = str;
+    POSITIVE('+'),
+
+    /**
+     * Represents the negative or reverse strand.
+     */
+    NEGATIVE('-'),
+
+    /**
+     * Denotes that a strand designation is not applicable
+     * or is unknown.
+     */
+    NONE('!');  // not really sure what we should do for the NONE Enum
+
+    /**
+     * Common alias for the {@link #POSITIVE} strand.
+     */
+    public static final Strand FORWARD = POSITIVE;
+
+    /**
+     * Common alias for the {@link #NEGATIVE} strand.
+     */
+    public static final Strand REVERSE = NEGATIVE;
+
+    /**
+     * Cached array of instances.
+     */
+    private final static Strand[] VALUES = values();
+
+    /**
+     * How we represent the strand as a single {@code char}.
+     */
+    private final char charEncoding;
+
+    /**
+     * How we represent the strand as a {@link String}.
+     */
+    private final String stringEncoding;
+
+    Strand(final char ch) {
+        charEncoding = ch;
+        stringEncoding = String.valueOf(charEncoding);
     }
 
     /**
      * provide a way to take an encoding string, and produce a Strand
      * @param encoding the encoding string
      * @return a Strand object, if an appropriate one cannot be located an IllegalArg exception
+     * @deprecated please use {@link #decode} instead.
+     */
+    @Deprecated
+    public static Strand toStrand(final String encoding) {
+        return decode(encoding);
+    }
+
+    /**
+     * Returns the {@link Strand} that a {@code char} value represents.
+     * @param ch the char encoding for a Strand.
+     * @throws IllegalArgumentException if input char does not
+     *   represent a {@link #Strand} instance.
+     * @return never {@code null}, a value so that {@code decode(c).encodeAsChar() == c}.
+     */
+    public static Strand decode(final char ch) {
+        for(final Strand value : VALUES) {
+            if (value.charEncoding == ch) {
+                return value;
+            }
+        }
+        throw new IllegalArgumentException("Unable to match encoding to Strand enum for encoding char: " + ch);
```

Comment:
.... so that is even suggesting yet another constant INVALID? this is a dangerous slippery slope. I think this even strengths the point that no more non-proper enum constant should be added... tomorrow someone may thing of something else like "NOT_APPLICABLE"... "BOTH".... In this particular case I think is best to simply fail early if an invalid string is to be converted to a Strand

Reply:
I am fine with the exception too, it was just an idea...